db/multi_cf_iterator_test.cc

```diff
@@ -0,0 +1,198 @@
+                    const std::vector<Slice>& expected_values) {
+    int i = 0;
+    MultiCfIterator* iter = db_->NewMultiCfIterator(ReadOptions(), cfhs);
+    for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
+      ASSERT_EQ(expected_keys[i], iter->key());
+      ASSERT_EQ(expected_values[i], iter->value());
+      ++i;
+    }
+    delete iter;
+    ASSERT_EQ(expected_keys.size(), i);
+  };
+
+  {
+    // Case 1: Unique key per CF
+    CreateAndReopenWithCF({"cf_1", "cf_2", "cf_3"}, options);
+
+    ASSERT_OK(Put(0, "key_1", "key_1_cf_0_val"));
+    ASSERT_OK(Put(1, "key_2", "key_2_cf_1_val"));
+    ASSERT_OK(Put(2, "key_3", "key_3_cf_2_val"));
+    ASSERT_OK(Put(3, "key_4", "key_4_cf_3_val"));
+
+    std::vector<Slice> expected_keys = {"key_1", "key_2", "key_3", "key_4"};
+    std::vector<Slice> expected_values = {"key_1_cf_0_val", "key_2_cf_1_val",
+                                          "key_3_cf_2_val", "key_4_cf_3_val"};
+
+    // Test for iteration over CF default->1->2->3
+    std::vector<ColumnFamilyHandle*> cfhs_order_0_1_2_3 = {
+        handles_[0], handles_[1], handles_[2], handles_[3]};
+    verify(cfhs_order_0_1_2_3, expected_keys, expected_values);
+
+    // Test for iteration over CF 3->1->default_cf->2
+    std::vector<ColumnFamilyHandle*> cfhs_order_3_1_0_2 = {
+        handles_[3], handles_[1], handles_[0], handles_[2]};
+    // Iteration order and the return values should be the same since keys are
+    // unique per CF
+    verify(cfhs_order_3_1_0_2, expected_keys, expected_values);
+  }
+  {
+    // Case 2: Same key in multiple CFs
+    options = CurrentOptions(options);
+    DestroyAndReopen(options);
+    CreateAndReopenWithCF({"cf_1", "cf_2", "cf_3"}, options);
+
+    ASSERT_OK(Put(0, "key_1", "key_1_cf_0_val"));
+    ASSERT_OK(Put(3, "key_1", "key_1_cf_3_val"));
+    ASSERT_OK(Put(1, "key_2", "key_2_cf_1_val"));
+    ASSERT_OK(Put(2, "key_2", "key_2_cf_2_val"));
+    ASSERT_OK(Put(0, "key_3", "key_3_cf_0_val"));
+    ASSERT_OK(Put(1, "key_3", "key_3_cf_1_val"));
+    ASSERT_OK(Put(3, "key_3", "key_3_cf_3_val"));
+
+    std::vector<Slice> expected_keys = {"key_1", "key_2", "key_3"};
+
+    // Test for iteration over CFs default->1->2->3
+    std::vector<ColumnFamilyHandle*> cfhs_order_0_1_2_3 = {
+        handles_[0], handles_[1], handles_[2], handles_[3]};
+    std::vector<Slice> expected_values = {"key_1_cf_0_val", "key_2_cf_1_val",
+                                          "key_3_cf_0_val"};
+    verify(cfhs_order_0_1_2_3, expected_keys, expected_values);
+
+    // Test for iteration over CFs 3->2->default_cf->1
+    std::vector<ColumnFamilyHandle*> cfhs_order_3_2_0_1 = {
+        handles_[3], handles_[2], handles_[0], handles_[1]};
+    expected_values = {"key_1_cf_3_val", "key_2_cf_2_val", "key_3_cf_3_val"};
+    verify(cfhs_order_3_2_0_1, expected_keys, expected_values);
+  }
+}
+
+TEST_F(MultiCfIteratorTest, DISABLED_IterateAttributeGroups) {
```

Comment:
Will re-enable it when `columns()` and `attribute_groups()` are implemented. Just to illustrate what they would return when implemented.